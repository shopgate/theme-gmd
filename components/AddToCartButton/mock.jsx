/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
