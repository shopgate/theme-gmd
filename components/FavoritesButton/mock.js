/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

const mockedProduct1 = {
  productId: 'foo',
};

const currentProduct = {
  productId: 'bar',
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateNotOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId],
    },
  },
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
      [currentProduct.productId]: {
        productData: currentProduct,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId, currentProduct.productId],
    },
  },
};

/**
 * Mocked state without data.
 * @type {{product: {currentProduct: {}, favorites: {}}}}
 */
export const mockedStateEmpty = {
  product: {
    currentProduct: {},
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
