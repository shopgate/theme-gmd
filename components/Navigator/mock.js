import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const router = {
  routing: false,
  stack: [
    {
      id: '44a34767-2683-4414-9de4-fd736808fdab',
      params: {},
      pathname: '/',
      pattern: '/',
      query: {},
      state: {},
    },
  ],
};

export const defaultState = {
  navigator: {
    backgroundColor: 'red',
    enabled: true,
    filterOpen: false,
    filterAttributeOpen: false,
    loginOpen: false,
    searchActive: false,
    showCartIcon: true,
    showIconShadow: false,
    showProgressBar: true,
    showSearch: true,
    showTitle: true,
    textColor: 'white',
    viewTracking: false,
  },
  search: {
    suggestion: {},
  },
  router,
  view: {
    isLoading: false,
  },
  ui: {
    general: {
      title: null,
    },
  },
};

/**
 * Creates store.
 * @param {Object} props Props which would override state.navigator props.
 * @return {Object}
 */
export const getStore = props => configureStore([thunk])({
  ...defaultState,
  navigator: {
    ...defaultState.navigator,
    ...props,
  },
  search: {
    suggestions: {},
  },
});
