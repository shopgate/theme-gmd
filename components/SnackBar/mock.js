/**
 * Copyright (c) 2017 - present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked toast state
 * @type {{toast: [null,null]}}
 */
export const mockedState = {
  toast: [
    {
      id: 1,
      message: 'Lorem ipsum dolor sit amet.',
      duration: 6000,
    },
  ],
};
