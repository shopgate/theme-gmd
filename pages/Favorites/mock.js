/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

