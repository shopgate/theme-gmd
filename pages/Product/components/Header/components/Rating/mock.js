/*
 *  Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export {
  getElementById,
};
