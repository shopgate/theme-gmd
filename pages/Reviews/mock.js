/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
