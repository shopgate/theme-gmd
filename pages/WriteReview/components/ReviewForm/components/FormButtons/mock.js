/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
};
