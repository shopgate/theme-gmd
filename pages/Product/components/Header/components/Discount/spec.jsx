import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { basicProductState } from '@shopgate/pwa-common-commerce/product/mock';

const mockedStore = configureStore();

describe('<Discount>', () => {
// eslint-disable-next-line require-jsdoc
  const createComponent = (state, id) => {
    const store = mockedStore(state);
    // eslint-disable-next-line global-require
    const Discount = require('./index').default;
    return mount(
      <Provider store={store}>
        <Discount productId={id} />
      </Provider>,
      mockRenderOptions
    );
  };

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render empty', () => {
    const { productId } = basicProductState.product.currentProduct;

    const cmp = createComponent(basicProductState, productId);

    expect(cmp).toMatchSnapshot();
    expect(cmp.find('Discount').children().length).toEqual(0);
  });

  it('should render discount', () => {
    const { productId } = basicProductState.product.currentProduct;
    const discount = 10;
    const state = {
      ...basicProductState,
      product: {
        ...basicProductState.product,
        productsById: {
          ...basicProductState.product.productsById,
          [productId]: {
            ...basicProductState.product.productsById[productId],
            productData: {
              ...basicProductState.product.productsById[productId].productData,
              price: {
                ...basicProductState.product.productsById[productId].productData.price,
                discount,
              },
            },
          },
        },
      },
    };

    const cmp = createComponent(state, productId);
    expect(cmp).toMatchSnapshot();

    expect(cmp.find('PlaceholderLabel').length).toEqual(1);
    expect(cmp.find('DiscountBadge').length).toEqual(1);
    expect(cmp.text()).toEqual(`-${discount}%`);
  });
});
