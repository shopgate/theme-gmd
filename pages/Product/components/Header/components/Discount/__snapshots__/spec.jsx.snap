// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Discount> should render discount 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Discount)
    productId="913"
  >
    <Discount
      dispatch={[Function]}
      price={
        Object {
          "currency": "EUR",
          "discount": 10,
          "info": "",
          "msrp": 0,
          "taxAmount": 0,
          "taxPercent": 0,
          "tiers": Array [],
          "totalPrice": 89,
          "unitPrice": 89,
          "unitPriceMax": 1000,
          "unitPriceMin": 0,
          "unitPriceNet": 89,
          "unitPriceStriked": 0,
          "unitPriceWithTax": 89,
        }
      }
      productId="913"
    >
      <PlaceholderLabel
        className="css-q720h0"
        ready={true}
        style={null}
      >
        <div
          className="css-1iex5li"
        >
          <DiscountBadge
            className=""
            discount={null}
            display="small"
            text="-10%"
          >
            <div
              data-test-id="-10%"
            >
              <Translate
                className="css-pzju2o "
                params={
                  Array [
                    null,
                  ]
                }
                string="-10%"
              >
                <span
                  className="css-pzju2o "
                >
                  -10%
                </span>
              </Translate>
            </div>
          </DiscountBadge>
        </div>
      </PlaceholderLabel>
    </Discount>
  </Connect(Discount)>
</Provider>
`;

exports[`<Discount> should render empty 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Discount)
    productId="913"
  >
    <Discount
      dispatch={[Function]}
      price={
        Object {
          "currency": "EUR",
          "info": "",
          "msrp": 0,
          "taxAmount": 0,
          "taxPercent": 0,
          "tiers": Array [],
          "totalPrice": 89,
          "unitPrice": 89,
          "unitPriceMax": 1000,
          "unitPriceMin": 0,
          "unitPriceNet": 89,
          "unitPriceStriked": 0,
          "unitPriceWithTax": 89,
        }
      }
      productId="913"
    />
  </Connect(Discount)>
</Provider>
`;
